packages/vitest/src/node/core.ts

```diff
@@ -592,6 +592,7 @@ export class Vitest {
       id = slash(id)
       updateLastChanged(id)
       if (await this.isTargetFile(id)) {
+        await this.globTestFiles([id])
```

Comment:
> And there is still a problem that this code will not even run if the file was added not in a core project.

Any pointers how I could reproduce this in the test case? I'm not sure if I yet understand the "core project" term in this new workspace feature context. 🙃 

https://github.com/vitest-dev/vitest/blob/c4c7720a55a5089124c44be53961bfc7320e133a/test/watch/test/workspaces.test.ts#L74

I'll look into those other points and remove usage of `globTestFiles`.

Reply:
Thanks for the explanation, that makes sense. The "root config" term is easier to understand, although "core project" might be better as `--config` option can indeed be used to pick non-root config as well. So using the `test/workspaces` as reference:

```
vitest/test/workspaces
├── space_1
|  └── vite.config.ts // <--- "Non-core project" config
|
├── space_2 // No project specific config here
|
├── space_3
|  └── vitest.config.ts // <--- "Non-core project" config
|
├── vitest.config.ts // <--- "Core project" config
└── vitest.workspace.ts
```

And I can reproduce the this issue now. When I add a test case that matches only the `test.includes` of `space_3/vitest.config.ts`, the `onAdd` is not run - exactly as you described. When `vitest run` is run again, this test case is included as initial `globFiles` recognizes it from the file system. I'll add test case for this too.

Reply:
Yes, exactly. For `space_2` there is still a config, but it's default Vite config since there is none on the file system. It doesn't extend the one in the root.

Reply:
> But do we still need to check file against `this.config` as well? That's the root/core config, right? Or does the root config cascade to project specific ones when resolved?

Where exactly do we need to check it? Project configs do not extend the root one.

Reply:
If root configuration specifies a custom `test.includes`:

```ts
include: [
  ...defaultInclude,
  '**/*.unconventional-pattern.ts',
]
```

And now we add a file that matches that pattern into a workspace project that has no configuration file. Does the workspace project use the `test.include` from root configuration or the Vite defaults, and simply ignores the added test case? Based on manual testing it seems that projects ignore that completely.

Reply:
Yes, it ignores it, unless the root config is specified as a project in `vitest.workspace.ts` file.

Reply:
The source of truth for what files to run are projects, not the root config. The root config will only affect what files are running if it is explicitly defined in the `vitest.workspace.ts` file.